client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java

```diff
@@ -475,6 +891,20 @@ public UpdateSettingsResponse putSettings(UpdateSettingsRequest updateSettingsRe
      * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html"> Update Indices Settings
      * API on elastic.co</a>
      */
+    public void putSettingsAsync(UpdateSettingsRequest updateSettingsRequest, RequestOptions options,
+                                 ActionListener<UpdateSettingsResponse> listener) {
+        restHighLevelClient.performRequestAsyncAndParseEntity(updateSettingsRequest, RequestConverters::indexPutSettings, options,
+                UpdateSettingsResponse::fromXContent, listener, emptySet());
+    }
+
+    /**
+     * Asynchronously updates specific index level settings using the Update Indices Settings API
+     * <p>
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html"> Update Indices Settings
+     * API on elastic.co</a>
+     * @deprecated Prefer {@link #putMappingAsync(PutMappingRequest, RequestOptions, ActionListener)}
```

Comment:
s/putMappingAsync/putSettingsAsync

Reply:
wow that's a good catch :) thanks for finding it.